src/history/hs_rec.c

```diff
@@ -197,6 +197,25 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
         cursor->set_key(cursor, 3, btree->id, key, tw->start_ts + 1);
         WT_ERR_NOTFOUND_OK(__wt_curhs_search_near_after(session, cursor), true);
     }
+
+    /*
+     * It is possible to insert a globally visible update into the history store with larger
+     * timestamps ahead of it. An example would be a mixed-mode update getting moved to the history
+     * store. This scenario can avoid detection earlier in reconciliation and result in an EBUSY
+     * being returned as it detects out-of-order timestamps. To prevent this we allow globally
+     * visible updates to fix history store content even if eviction is running concurrently with a
+     * checkpoint.
+     *
+     * This is safe because global visibility considers the checkpoint transaction id and timestamp
+     * while it is running, i.e. if the update is globally visible to eviction it will be globally
+     * visible to checkpoint and the modifications it makes to the history store will be the same as
+     * what checkpoint would've done.
+     */
+    if (error_on_ooo_ts &&
+      (__wt_txn_tw_stop_visible_all(session, tw) || __wt_txn_tw_start_visible_all(session, tw))) {
```

Comment:
I don't clearly understand the reason behind the need for `stop_visible_all`. If the stop is visible to everyone there is no need to insert this update to the history store. The start should be enough.

Reply:
I reverted the stop change.